src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -941,6 +943,40 @@ public static RestoreAuditProperties GetRestoreAuditProperties(IMSBuildItem spec
             return null;
         }
 
+        public static NuGetVersion GetSdkAnalysisLevel(string sdkAnalysisLevel)
+        {
+            if (!string.IsNullOrEmpty(sdkAnalysisLevel))
+            {
+                if (NuGetVersion.TryParse(sdkAnalysisLevel, out NuGetVersion version))
+                {
+                    return version;
+                }
+                else
+                {
+                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Invalid_AttributeValue, "SdkAnalysisLevel", sdkAnalysisLevel, "9.0.100"));
+                }
+            }
+
+            return null;
+        }
```

Comment:
I find this to be a better pattern where the method exits early and has less nesting:

```suggestion
        public static NuGetVersion GetSdkAnalysisLevel(string sdkAnalysisLevel)
        {
            if (string.IsNullOrEmpty(sdkAnalysisLevel))
            {
                return null;
            }
            
            if (!NuGetVersion.TryParse(sdkAnalysisLevel, out NuGetVersion version))
            {
                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Invalid_AttributeValue, "SdkAnalysisLevel", sdkAnalysisLevel, "9.0.100"));
            }
            
            return version;
        }
```